File: adds-on/01-Data Preprocessing/Pre_Data.py
# Importing the libraries
import pandas as pd


# Step 1 : Importing the dataset
print("Begin Step 1 : Importing the dataset:")
path = "K:/"
csv_file = "data.csv"
csv_file_path = path+ "/"+ csv_file
dataset = pd.read_csv(csv_file_path)
print(dataset)
print("End Step 1 : Importing the dataset: \"", csv_file, "\"")
print()

# Step 2 : Check out the Missing Values
print("Begin Step 2 : After dropping the Missing Values")
import numpy
dataset = dataset.replace(0, numpy.NaN)
# fill missing values with mean column values
dataset.fillna(dataset.median(), inplace=True)
print(dataset)
print("End Step 2 : After dropping the Missing Values")
print()

# Step 3 : Encoding categorical data
print("Begin Step 3 : Encoding categorical data")
State = {'New York': 1,'California': 2, 'Florida': 3}
dataset.State = [State[item] for item in dataset.State]
y = dataset.iloc[:, 4].values
print(dataset)
print("End Step 3 : Encoding categorical data")
print()

# Step 4 :  Feature Scaling
print("Begin Step 4 : Feature Scaling")
from sklearn.preprocessing import MinMaxScaler
scaler = MinMaxScaler()
dataset = scaler.fit_transform(dataset)
print(dataset)
print("End Step 4 : Feature Scaling")
print()

# Step 5 : Splitting the dataset into the Training set and Test set
from sklearn.model_selection import train_test_split
X = dataset
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)
print("Step 5 : Splitting the dataset into the Training set and Test set")
print("X_train")
print(X_train)
print()
print("X_test")
print(X_test)
print()
print("y_train")
print(y_train)
print()
print("y_test")
print(y_test)
print()

File: adds-on/01-Data Preprocessing/Pre_Data.R
# Importing the dataset
setwd("K:/")
dataset = read.csv('data.csv')
print(dataset)

# Taking care of missing data
dataset$Age = ifelse(is.na(dataset$Age),
                     ave(dataset$Age, FUN = function(x) mean(x, na.rm = TRUE)),
                     dataset$Age)
dataset$Salary = ifelse(is.na(dataset$Salary),
                        ave(dataset$Salary, FUN = function(x) mean(x, na.rm = TRUE)),
                        dataset$Salary)
print("Taking care of missing data")
print(dataset)

# Encoding categorical data
dataset$Country = factor(dataset$Country,
                         levels = c('France', 'Spain', 'Germany'),
                         labels = c(1, 2, 3))
dataset$Purchased = factor(dataset$Purchased,
                           levels = c('No', 'Yes'),
                           labels = c(0, 1))
print("Encoding categorical data")
print(dataset)

# Splitting the dataset into the Training set and Test set
#install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(dataset$Country, SplitRatio = 0.8)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)
print("Splitting the dataset into the Training set and Test set")
print(training_set)
print(test_set)

# Feature Scaling
#training_set = as.double(training_set)
#test_set = as.double(test_set)

training_set[, 2:3] = scale(training_set[, 2:3])
test_set[, 2:3] = scale(test_set[, 2:3])
print("Feature Scaling")
print(training_set)
print(test_set)